### PR TITLE
Slack engine fix results

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?

Slack engine fix results and make them easier to deal with.
### Previous Behavior

Results would be sent directly into the channel via a message, this was problematic as there is a character limit and the formatting wasn't correct.
### New Behavior

Results are now attached as a file in the channel.

### CODE REVIEW

**Comment by cachedout** [comment]
Go Go Jenkins!

File: salt/master.py [hunk 1]
```diff
@@ -540,7 +540,7 @@ def start(self):
 
             if 'reactor' in self.opts and 'reactor' not in self.opts['engines']:
                 log.info('Enabling the reactor engine')
-                self.opts['engines']['reactor'] = {}
+                self.opts['engines'].append({'reactor': {}})
```
Comment (cachedout) [inline salt/master.py]: Are you sure this change is right? It doesn't look like the master is starting correctly with it.

Comment (garethgreenaway) [inline salt/master.py]: It was working for me. I'll take another look.

**Comment by garethgreenaway** [comment]
Tests failed but unrelated to my change.

**Comment by cachedout** [comment]
Thanks very much @garethgreenaway

**Pull request merged**